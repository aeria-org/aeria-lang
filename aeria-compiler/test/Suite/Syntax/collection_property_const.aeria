collection ConstProperty {
  properties {
    constStr        const @value("test")
    constNum        const @value(1.1)
    constInt        const @value(1)
    constBool       const @value(false)
    constUndefined  const @value(undefined)
    constNull       const @value(null)
  }
}
