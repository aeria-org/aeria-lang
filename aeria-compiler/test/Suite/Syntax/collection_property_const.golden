{
  "kind": "Program",
  "collections": [
    {
      "properties": [
        {
          "type": {
            "kind": "PConst"
          },
          "name": {
            "value": "prop1",
            "kind": "PropertyName"
          },
          "kind": "Property",
          "attributes": [
            {
              "kind": "Attribute",
              "attributeValue": {
                "literal": {
                  "value": "test",
                  "kind": "LString"
                },
                "kind": "ALiteral"
              },
              "attributeName": {
                "value": "value",
                "kind": "AttributeName"
              }
            }
          ]
        },
        {
          "type": {
            "kind": "PConst"
          },
          "name": {
            "value": "prop2",
            "kind": "PropertyName"
          },
          "kind": "Property",
          "attributes": [
            {
              "kind": "Attribute",
              "attributeValue": {
                "literal": {
                  "value": 1.1,
                  "kind": "LNum"
                },
                "kind": "ALiteral"
              },
              "attributeName": {
                "value": "value",
                "kind": "AttributeName"
              }
            }
          ]
        },
        {
          "type": {
            "kind": "PConst"
          },
          "name": {
            "value": "prop3",
            "kind": "PropertyName"
          },
          "kind": "Property",
          "attributes": [
            {
              "kind": "Attribute",
              "attributeValue": {
                "literal": {
                  "value": 1,
                  "kind": "LInteger"
                },
                "kind": "ALiteral"
              },
              "attributeName": {
                "value": "value",
                "kind": "AttributeName"
              }
            }
          ]
        },
        {
          "type": {
            "kind": "PConst"
          },
          "name": {
            "value": "prop4",
            "kind": "PropertyName"
          },
          "kind": "Property",
          "attributes": [
            {
              "kind": "Attribute",
              "attributeValue": {
                "literal": {
                  "value": false,
                  "kind": "LBoolean"
                },
                "kind": "ALiteral"
              },
              "attributeName": {
                "value": "value",
                "kind": "AttributeName"
              }
            }
          ]
        },
        {
          "type": {
            "kind": "PConst"
          },
          "name": {
            "value": "prop5",
            "kind": "PropertyName"
          },
          "kind": "Property",
          "attributes": [
            {
              "kind": "Attribute",
              "attributeValue": {
                "literal": {
                  "kind": "LUndefined"
                },
                "kind": "ALiteral"
              },
              "attributeName": {
                "value": "value",
                "kind": "AttributeName"
              }
            }
          ]
        },
        {
          "type": {
            "kind": "PConst"
          },
          "name": {
            "value": "prop6",
            "kind": "PropertyName"
          },
          "kind": "Property",
          "attributes": [
            {
              "kind": "Attribute",
              "attributeValue": {
                "literal": {
                  "kind": "LNull"
                },
                "kind": "ALiteral"
              },
              "attributeName": {
                "value": "value",
                "kind": "AttributeName"
              }
            }
          ]
        }
      ],
      "name": {
        "value": "collectionpropertyconsttest",
        "kind": "CollectionName"
      },
      "kind": "Collection"
    }
  ]
}
